### PR TITLE
Add missing xml docs for all file provider implementations

### PR DESCRIPTION
**Pull request by natemcmaster**
Add documentation for the public API surface. Includes documentation on hidden behaviors, such as the 'DOTNET_USE_POLLING_FILE_WATCHER' environment variable

cc @troydai

### CODE REVIEW

File: src/Microsoft.Extensions.FileProviders.Composite/CompositeDirectoryContents.cs [hunk 1]
```diff
@@ -73,6 +73,10 @@ private void EnsureFilesAreInitialized()
             }
         }
 
+        /// <summary>
+        /// Creates an enumerator for all files in all providers given. Does not duplicate items, even if multiple providers for the same file exist.
```
Comment (troydai) [inline src/Microsoft.Extensions.FileProviders.Composite/CompositeDirectoryContents.cs]: Break the line earlier to improve readability

**Comment by troydai** [comment]
Looks good to me

**Comment by natemcmaster** [comment]
@troydai is that signoff?

File: src/Microsoft.Extensions.FileProviders.Embedded/EmbeddedFileProvider.cs [hunk 2]
```diff
@@ -165,6 +173,11 @@ public IDirectoryContents GetDirectoryContents(string subpath)
             return new EnumerableDirectoryContents(entries);
         }
 
+        /// <summary>
+        /// Embedded files do not change.
+        /// </summary>
+        /// <param name="pattern"></param>
```
Comment (pranavkm) [inline src/Microsoft.Extensions.FileProviders.Embedded/EmbeddedFileProvider.cs]: > The pattern to watch.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**